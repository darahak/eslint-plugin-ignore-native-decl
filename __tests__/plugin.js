const path = require('path');
const fs = require('fs');
const processor = require('../lib/processors/nativeFunction');

function getFixturePath(file) {
  return path.join(__dirname, 'fixtures', file);
}

function format(file) {
  const text = fs.readFileSync(getFixturePath(file), { encoding: 'utf-8' });
  return processor.preprocess(text);
}

test('Replaces all occurrences of native functions with empty functions', () => {
  const transformed = format('nativeDecl.js')[0];

  expect(transformed).not.toContain('native function');
  expect(transformed).toContain('function() {}');

  expect(transformed).toMatchSnapshot();
});

test('Leaves source code untouched if it has no native function declarations', () => {
  const fixture = 'standardDecl.js';
  const transformed = format(fixture)[0];

  expect(transformed).not.toContain('native function');

  const original = fs.readFileSync(getFixturePath(fixture), {
    encoding: 'utf-8'
  });

  expect(transformed).toEqual(original);

  expect(transformed).toMatchSnapshot();
});

// TODO: Write a test for postprocess().
