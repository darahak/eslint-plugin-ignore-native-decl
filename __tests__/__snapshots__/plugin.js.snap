// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leaves source code untouched if it has no native function declarations 1`] = `
"function a() {}
var b = function() {};
let c = function() {};
const d = function() {};
const e = () => {};
"
`;

exports[`Replaces all occurrences of native functions with empty functions 1`] = `
"var aaa = function() {};
var bbb = function() {};
var ccc = function() {};
"
`;
